src/components/Header/components/Login.tsx

```diff
@@ -1,61 +1,110 @@
-import React from "react";
-import { chakra } from "@chakra-ui/react";
+import React, { useEffect, useRef } from "react";
+import { Box, chakra, VStack } from "@chakra-ui/react";
 import FocusLock from "@chakra-ui/focus-lock";
 
 import Button from "../../Button/Button";
 import Icon from "../../Icons/Icon";
 import Link from "../../Link/Link";
 import List from "../../List/List";
+import Text from "../../Text/Text";
 import { LoginProps } from "./UpperNav";
 
-const Login = chakra(({ loginOpen, setLoginOpen }: LoginProps) => {
+const Login = chakra(({ loginOpen, patronName, setLoginOpen }: LoginProps) => {
+  const greeting = useRef(null);
+
+  useEffect(() => {
+    if (patronName && loginOpen) {
+      greeting.current.focus();
+    }
+  }, [loginOpen, patronName]);
+
   return (
     <FocusLock isDisabled={!loginOpen}>
       <Button
         buttonType="link"
-        className={`withIcon ${loginOpen ? "closeLoginButton" : "loginButton"}`}
+        className={`withIcon ${
+          loginOpen
+            ? "closeLoginButton"
+            : patronName
+            ? "accountButton"
+            : "loginButton"
+        }`}
         id="loginButton"
         onClick={() => setLoginOpen(!loginOpen)}
       >
-        {loginOpen ? "Close" : "Log In"}
+        {loginOpen ? "Close" : patronName ? "My Account" : "Log In"}
         <Icon align="right" name={loginOpen ? "close" : "arrow"} size="small" />
       </Button>
       {loginOpen && (
-        <List
-          id="login"
-          className="loginMenu"
-          noStyling
-          type="ul"
-          listItems={[
-            <Link id="logInCatalog" key="logInCatalog" href="#" type="button">
-              <Icon
-                align="left"
-                color="ui.white"
-                id="log-in-catalog-icon"
-                name="utilityAccountFilled"
-                size="medium"
-                title="Log in to your account"
-              />
-              Log Into The Catalog
-            </Link>,
+        <VStack className="loginMenu">
+          {patronName && (
+            <Box
+              id="patronGreeting"
+              data-testid="patronGreeting"
+              ref={greeting}
+              tabIndex={-1}
```

Comment:
Set this to `{0}`. This is similar to what's in prod: https://github.com/NYPL/dgx-header-component/blob/master/src/components/MyNypl/MyNypl.jsx#L83

The `-1` is correct and allows us to programmatically focus on the element, but we also want to tab on this element within the focus trap.